spec/lib/base16_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe Base16 do
+
+  context 'with reasonable inputs' do
+
+    context 'given "Hello, World"' do
+      let(:input) { 'Hello, world' }
+
+      it 'returns a known value' do
+        encoded = described_class.encode16(input)
+        # The IRS confirmed this value:
+        expect(encoded).to eq '48656C6C6F2C20776F726C64'
+
+        decoded = described_class.decode16(encoded)
+        expect(decoded).to eq input
+      end
+
+      it 'returns a value with uppercase letters' do
+        encoded = described_class.encode16(input)
+        expect(encoded).to eq(encoded.upcase)
+      end
+    end
+
+    context 'given a sequence of zeroes' do
+      let(:input) { "\x00" }
+      it 'does not truncate them' do
+        encoded = described_class.encode16(input)
+        expect(encoded).to eq '00'
+
+        decoded = described_class.decode16(encoded)
+        expect(decoded).to eq(input)
+      end
+    end
+  end
+
+  context 'with a less reasonable input', skip: true do
+    context 'given a zany-face emoji' do
+      let(:input) { "🤪" }
```

Comment:
I'm skipping this test (which won't pass) but including it for illustration. I think out given input will always be a series of bytes and faithfully preserving emoji isn't a requirement, _but_ I want to highlight that this limitation is there.

Reply:
This was fixed by Zach's PR to this PR. 😄